packages/react-server-dom-webpack/src/ReactFlightWebpackNodeLoader.js

```diff
@@ -70,19 +70,23 @@ export async function resolve(
       );
     }
   }
-  // We intentionally check the specifier here instead of the resolved file.
-  // This allows package exports to configure non-server aliases that resolve to server files
-  // depending on environment. It's probably a bad idea to export a server file as "main" though.
-  if (specifier.endsWith('.server.js')) {
+  const resolved = defaultResolve(specifier, context, defaultResolve);
```

Comment:
Here's something funky. The actual `defaultResolve` is sync. No promise. Is it supposed to be chainable, i.e. _can_ this return a Promise?

Reply:
That's weird. You'd expect they'd want to do that asynchronously.

But it's supposed to be chainable so you can do stuff like this:

https://github.com/facebook/react/blob/master/fixtures/flight/loader/index.js#L30